modin/experimental/core/execution/native/implementations/omnisci_on_native/dataframe/dataframe.py

```diff
@@ -335,7 +335,13 @@ def mask(
 
         if row_labels is not None or row_positions is not None:
             op = MaskNode(base, row_labels=row_labels, row_positions=row_positions)
+            if row_positions is not None:
+                idx = self.index
```

Comment:
Access to the index triggers frame execution. The goal of this backend is to provide lazy execution, which means you don't have a pre-computed index for each frame. Why do you need to compute the index here?